stan/math/prim/functor/broadcast_array.hpp

```diff
@@ -23,14 +24,11 @@ class broadcast_array {
 
   /** \ingroup type_trait
    * We can assign any right hand side which allows for indexing to a
-   * broadcast_array. The idea is that the entry for the first index is what
-   * gets assigned. The most common use-case should be where the rhs is some
-   * container of length 1.
+   * broadcast_array. The idea is that the argument is summed.
    */
   template <typename Y>
   void operator=(const Y& m) {
-    ref_type_t<Y> m_ref = m;
-    prim_ = m_ref[0];
+    prim_ = sum(m);
```

Comment:
[question]
Does this cause any extra work if `m` is not a container?

Reply:
For non-container `sum` is a no-op. So assuming compiler is smart enough to inline the call, no.